glommio/src/lib.rs

```diff
@@ -106,89 +127,82 @@
-//!    Because `io_uring` allows for polling in the ring file descriptor,
-//!    it is safe to `park()` even if work is present in the latency ring:
-//!    before going to sleep, the latency ring's file descriptor is
-//!    registered with the main ring and any events it sees will also wake
-//!    up the main ring.
+//! ## `io_uring`
+//!
+//! This crate depends heavily on Linux's `io_uring`. The reactor will register
+//! 3 rings per CPU:
+//!
+//!  * *Main ring*: The main ring, as its name implies, is where most operations
+//!    will be placed. Once the reactor is parked, it only returns if the main
+//!    ring has events to report.
+//!
+//!  * *Latency ring*: Operations that are latency sensitive can be put in the
+//!    latency ring. The crate has a function called `yield_if_needed()` that
+//!    efficiently checks if there are events pending in the latency ring.
+//!    Because this crate uses `cooperative` programming, tasks run until they
+//!    either complete or decide to yield, which means they can run for a very
+//!    long time before tasks that are latency sensitive have a chance to run.
+//!    Every time you fire a long-running operation (usually a loop) it is good
+//!    practice to check `yield_if_needed()` periodically (for example after x
+//!    iterations of the loop). In particular, a when a new priority class is
+//!    registered, one can specify if it contains latency sensitive tasks or
+//!    not. And if the queue is marked as latency sensitive, the Latency enum
+//!    takes a duration parameter that determines for how long other tasks can
+//!    run even if there are no external events (by registering a timer with the
+//!    io_uring). If no runnable tasks in the system are latency sensitive, this
+//!    timer is not registered. Because `io_uring` allows for polling in the
+//!    ring file descriptor, it is safe to `park()` even if work is present in
+//!    the latency ring: before going to sleep, the latency ring's file
+//!    descriptor is registered with the main ring and any events it sees will
+//!    also wake up the main ring.
 //!
 //!  * *Poll ring*: Read and write operations on NVMe devices are put in the
-//!    poll ring. The poll ring does not rely on interrupts so the system
-//!    has to keep constantly polling if there is any pending work. By
-//!    not relying on interrupts we can be even more efficient with I/O in
-//!    high IOPS scenarios
+//!    poll ring. The poll ring does not rely on interrupts so the system has to
+//!    keep constantly polling if there is any pending work. By not relying on
+//!    interrupts we can be even more efficient with I/O in high IOPS scenarios
 //!
 //! ## Before using Glommio
 //!
-//! Please note Glommio requires at least 256 KiB of locked memory for `io_uring`
-//! to work. You can increase the `memlock` resource limit (rlimit) as follows:
+//! Please note Glommio requires at least 256 KiB of locked memory for
```

Comment:
This is a pre-existing problem, but this is a mistake. The minimum requirement currently sits at 512kB.

Reply:
See that in the instructions below we actually say 512...